GVRf/Framework/src/org/gearvrf/scene_objects/GVRVideoSceneObject.java

```diff
@@ -167,8 +189,16 @@ public MediaPlayer getMediaPlayer() {
      * @param mediaPlayer
      *            An Android {@link MediaPlayer}
      */
-    public void setMediaPlayer(MediaPlayer mediaPlayer) {
-        mVideo.setMediaPlayer(mediaPlayer);
+    public void setMediaPlayer(final MediaPlayer mediaPlayer) {
+        if (mVideo == null) {
```

Comment:
I guess the advantage of having the caller provide the media player gives it control over its life cycle. Having this new constructor breaks that. Ignore my question.